### PR TITLE
:sparkles: Add support to load gifs in button animation

### PR DESCRIPTION
**Pull request by mdhanif-simformsolutions**
- Add support to load gifs in button animation
- Added a demo sample button with gif animation

### CODE REVIEW

File: ssjetpackcomposeprogressbutton/build.gradle [hunk 1]
```diff
@@ -56,6 +56,8 @@ android {
 dependencies {
     implementation(libs.ui)
     implementation(libs.material3)
+    implementation(libs.coil.compose)
+    implementation libs.coil.gif
```
Comment (yashwantgowla-simform) [inline ssjetpackcomposeprogressbutton/build.gradle:60]: Add round brackets

**Review by yashwantgowla-simform — CHANGES_REQUESTED** [review]
Changes requested

File: ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt [hunk 2]
```diff
@@ -97,7 +102,9 @@ fun SSJetPackComposeProgressButton(
     onClick: () -> Unit,
     assetColor: Color,
     buttonState: SSButtonState,
-    buttonBorderStroke: BorderStroke? = null,
+    buttonBorderWidth: Dp? = 0.dp,
+    buttonBroderColor: Color? = null,
```
Comment (yashwantgowla-simform) [inline ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt:106]: type error: buttonBorderColor

File: ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt [hunk 3]
```diff
@@ -54,15 +57,17 @@ import kotlinx.coroutines.delay
  * @param onClick Will be called when the user clicks the button.
  * @param assetColor Color to be applied to icon and text inside button.
  * @param buttonState represent the state of button from IDLE, LOADING, SUCCESS, FAILIURE from SSButtonState.
- * @param buttonBorderStroke Border to draw around the button.
+ * @param buttonBorderWidth Border width to draw around the button.
+ * @param buttonBroderColor Border color applied to button.
```
Comment (yashwantgowla-simform) [inline ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt:61]: typo error

File: ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt [hunk 4]
```diff
@@ -135,6 +142,17 @@ fun SSJetPackComposeProgressButton(
     var failureAlphaValue by remember { mutableStateOf(zeroFloat) }
     var progressAlphaValue by remember { mutableStateOf(zeroFloat) }
     var cornerRadiusValue by remember { mutableStateOf(cornerRadius) }
+    val borderColor by animateColorAsState(
+        targetValue = (
+                if (buttonState == SSButtonState.LOADING) animatedButtonBorderColor
+                else buttonBroderColor
+                ) ?: Color.Transparent,
```
Comment (yashwantgowla-simform) [inline ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt:149]: Keep it like this for readability:

if (buttonState == SSButtonState.LOADING)
                    animatedButtonBorderColor
                else
                    buttonBroderColor
                ) ?: Color.Transparent,

File: ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt [hunk 5]
```diff
@@ -242,7 +260,10 @@ fun SSJetPackComposeProgressButton(
             enabled = enabled && buttonState != SSButtonState.LOADING,
             elevation = elevation,
             shape = RoundedCornerShape(ssAnimateIntAsState(cornerRadiusValue, speedMillis)),
-            border = buttonBorderStroke,
+            border = BorderStroke(
+                width = buttonBorderWidth ?: 0.dp,
```
Comment (yashwantgowla-simform) [inline ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt:264]: Replace with ZERO.dp

File: ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt [hunk 6]
```diff
@@ -97,7 +102,9 @@ fun SSJetPackComposeProgressButton(
     onClick: () -> Unit,
     assetColor: Color,
     buttonState: SSButtonState,
-    buttonBorderStroke: BorderStroke? = null,
+    buttonBorderWidth: Dp? = 0.dp,
```
Comment (yashwantgowla-simform) [inline ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt:105]: ZERO.dp

File: ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt (hunk 5 above)
Comment (mdhanif-simformsolutions) [reply ssjetpackcomposeprogressbutton/src/main/java/com/simform/ssjetpackcomposeprogressbuttonlibrary/SSJetPackComposeProgressButton.kt:264]: We will be removing ZERO.dp in the upcoming PR.

**Review by yashwantgowla-simform — APPROVED** [review]

**Pull request merged**